descartes_planner/src/planning_graph.cpp

```diff
@@ -595,39 +595,81 @@ bool PlanningGraph::removeTrajectory(TrajectoryPtPtr point)
 
 bool PlanningGraph::findStartVertices(std::list<JointGraph::vertex_descriptor> &start_points)
 {
-  std::pair<VertexIterator, VertexIterator> vi = boost::vertices(dg_);
-  for (VertexIterator vert_iter = vi.first; vert_iter != vi.second; ++vert_iter)
-  {
-    JointGraph::vertex_descriptor jv = *vert_iter;
+  // Create local id->vertex map  
+  std::map<TrajectoryPt::ID, JointGraph::vertex_descriptor> joint_vertex_map;
+  int num_joints = recalculateJointSolutionsVertexMap(joint_vertex_map);
+
+  // Find the TrajectoryPt ID of the first point specified by the user
+  TrajectoryPt::ID cart_id = generate_nil();
 
-    std::pair<InEdgeIterator, InEdgeIterator> in_ei = boost::in_edges(jv, dg_);
-    if (in_ei.first == in_ei.second)
+  for(std::map<TrajectoryPt::ID, CartesianPointInformation>::iterator c_iter = cartesian_point_link_->begin();
+      c_iter != cartesian_point_link_->end(); c_iter++)
+  {
+    if(c_iter->second.links_.id_previous.is_nil())
     {
-      // debug
-      ROS_DEBUG("Graph start node: %d", (int)jv);
-      start_points.push_back(jv);
+      cart_id = c_iter->first;
+      break;
     }
   }
+
+  // Test for error case
+  if (cart_id.is_nil())
+  {
+    ROS_ERROR("Could not locate TrajectoryPt with nil previous point. Graph may be cyclic.");
```

Comment:
Should this ever happen?  Maybe a comment to describe when it does (if it could)?